lib/io.c

```diff
@@ -39,7 +39,7 @@ int metal_io_block_read(struct metal_io_region *io, unsigned long offset,
 	unsigned char *dest = dst;
 	int retlen;
 
-	if (offset >= io->size)
+	if (!ptr || offset >= io->size)
```

Comment:
(offset >= io->size) is already tested in metal_io_virt 
So testing !ptr should be sufficient

Reply:
Agree. I'll update this patch to remove it as you suggested.